core/src/main/scala/chisel3/Bits.scala

```diff
@@ -31,12 +31,6 @@ private[chisel3] sealed trait ToBoolable extends Element {
     */
   final def asBool: Bool = macro SourceInfoWhiteboxTransform.noArg
 
-  @deprecated(
```

Comment:
won't removing this suddenly allow people using Scala 2 (i.e. everyone) to call `asBool()` without a warning again?

Reply:
I don't think so. Try putting parens on an asBool in a test and it will be a compile error. The error perhaps is not so good as it says
```
overloaded method value apply with alternatives:
  (x: BigInt,y: BigInt)chisel3.UInt <and>
  (x: Int,y: Int)chisel3.UInt <and>
  (x: chisel3.UInt)chisel3.Bool <and>
  (x: Int)chisel3.Bool <and>
  (x: BigInt)chisel3.Bool
 cannot be applied to ()
  out := a.asBool()
```
@jackkoenig is there a way to make that better?

Reply:
oh nice, somehow I thought that you could always add the paranthesis

Reply:
> oh nice, somehow I thought that you could always add the parentheses

If the method is defined with `()`, you could call with or without. If defined without, you always have had to call without. Obviously this "either or" was confusing so they changed it in Scala 3 and it's noisily deprecated in 2.13 .

> @jackkoenig is there a way to make that better?

We could do a def macro similarly to the error/warning for `5.U(8)`, but not sure if it's worth it.